tests/components/update/test_init.py

```diff
@@ -86,7 +87,12 @@ async def test_update(hass: HomeAssistant) -> None:
     update._attr_latest_version = None
     assert update.state is None
 
+    # Test entity category becomes config when its possible to install
+    update._attr_supported_features = UpdateEntityFeature.INSTALL
+    assert update.entity_category is EntityCategory.CONFIG
+
     # UpdateEntityDescription was set
+    update._attr_supported_features = 0
     update.entity_description = UpdateEntityDescription(key="F5 - Its very refreshing")
     assert update.device_class is None
     assert update.entity_category is EntityCategory.CONFIG
```

Comment:
How is this True?
If `update._attr_supported_features = 0` (L95), this should be DIAGNOSTICS?

Reply:
Because of
https://github.com/home-assistant/core/blob/073fb40b79cf8aa06790fdceb23b6857db888c99/homeassistant/components/update/__init__.py#L149

Reply:
So this is where things get nasty I guess...

The UpdateEntityDescription defaults to CONFIG, we can't make that dynamic

Reply:
If you use `UpdateEntityDescription` you do that to get a stable simple interface to "control" the entity, having CONFIG as the default there I think is OK, and when you do not use `UpdateEntityDescription` you get this magic.